cowprotocol/cowswap
# feat(widget): use custom token lists in widget

# Summary

In this PR I wired up [the UI](https://github.com/cowprotocol/cowswap/pull/3390) with CoW Swap app.

1. New parameters in the widget config:

```
  /**
   * When enabled, the widget will only use the token lists provided in the `tokenLists` property.
   * Otherwise, it will append the provided token lists to default ones.
   */
  useOnlyProvidedTokenLists: boolean
  /**
   * The token lists to use in the widget
   */
  tokenLists: TokenList[]
```

2. @fairlighteth I'm sorry, but I deleted `name` parameter from token list model, because it's artificial. To get a real list name we should fetch the list source and pick the name from it. I think this is our of the widget configurator scope goals.

>`useOnlyProvidedTokenLists` is disabled by default! You can only test it if you enable it in the code



<img width="293" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/7104f0ae-155b-42e0-8d85-91f6be699b52">

  # To Test

1. Add some token lists in configurator
- [ ] the list is displayed in CoW Swap token lists
2. Remove some token lists in configurator
- [ ] the list is removed in CoW Swap token lists
4. Open the widget host just as an App
- [ ] the app shouldn't display lists that were added in widget mode
